### PR TITLE
Reduce instruction size

### PR DESCRIPTION
**Pull request by take-cheeze**
Implementation of #4150 .

### CODE REVIEW

**Comment by take-cheeze** [comment]
Implementation should be done.

**Comment by matz** [comment]
`mrb_reserved_symbols` in `symbol.h` seems like premature optimization. They will be overridden when embedded symbols come. Plain `mrb_intern` should be enough for the time being.

**Comment by take-cheeze** [comment]
Even if embedded symbols come, I think these reserved symbols is needed for symbols that is used in language feature.
Also they could be replaced with macro when its optimization doesn't match.

And code generator could be optimized with it too.
(I didn't modified it this time.)

File: mrbgems/mruby-compiler/core/codegen.c [hunk 1]
```diff
@@ -982,7 +983,9 @@ gen_call(codegen_scope *s, node *tree, mrb_sym name, int sp, int val, int safe)
     gen_move(s, cursp(), recv, 1);
     skip = genjmp2(s, OP_JMPNIL, cursp(), 0, val);
   }
-  idx = new_sym(s, sym);
+  if (!mrb_symbol_constsym_send_p(sym)) {
```
Comment (take-cheeze) [inline mrbgems/mruby-compiler/core/codegen.c]: Need to check `&& n == 1`.

**Pull request merged**